static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -772,6 +779,17 @@ class SpanTreeModel {
             false
           );
 
+          // We need to adjust the timestamps for this embedded transaction only if it is not within the bounds of its parent span
+          if (
+            parsedRootSpan.span.start_timestamp < this.span.start_timestamp ||
+            parsedRootSpan.span.timestamp > this.span.timestamp
+          ) {
+            const startTimeDelta =
+              this.span.start_timestamp - parsedRootSpan.span.start_timestamp;
+            adjustEmbeddedTransactionTimestamps(parsedRootSpan, startTimeDelta);
```

Comment:
I wonder if there is a way to make `adjustEmbeddedTransactionTimestamps` return a modified `root` and then `parsedRootSpan` can be updated in this block of code. When first looking at `adjustEmbeddedTransactionTimestamps` I wouldn't know that it overwrites its parameter, it's an implicit mutating behavior and is easy to overlook. We can make this more obvious by returning a new root. What do you think?

Reply:
@udameli It should be trivial to add this, but I'm not too sure how returning the root will make this behaviour clear. 
Since the function is recursively traversing the root and all of its descendants, it seems a bit strange to me for this function to return the same root after it and all its descendants are modified. 

In my mind, returning a root would imply that a new tree was built, rather than the existing tree having been modified. Am I misunderstanding or missing something here?

Reply:
generally, it's more common to keep functions [pure](https://www.freecodecamp.org/news/what-is-a-pure-function-in-javascript-acb887375dfe/). When functions start mutating their arguments, it's not immediately obvious and can make it hard to keep track of changes to data.

in this example, something like this could be an alternative (I didn't test the code):
```
export function adjustEmbeddedTransactionTimestamps(
  root: SpanTreeModel,
  startTimeDelta: number
) {
  const adjustedRoot = root;
  adjustedRoot.span.start_timestamp += startTimeDelta;
  adjustedRoot.span.timestamp += startTimeDelta;

  adjustedChildren = adjustedRoot.children.map(child => {
    const adjustedChild = adjustEmbeddedTransactionTimestamps(child, startTimeDelta);
    return adjustedChild;
   }

  adjustedRoot.children = adjustedChildren;
  return adjustedRoot;
  );
}
```

that way input is never mutated. but... this isn't that much more readable now that I think about it lol. maybe keep the function as it is and add a comment that it mutates `root`? choosing between two options, mutating does seem like a better alternative :)

Reply:
Yeah, I think it would just overcomplicate the code here imo. I agree with the concept of keeping functions pure when applying functional programming principles, but `SpanTreeModel` is an object, and we don't seem to be following any other functional programming practices in here. `parsedRootSpan` is also only ever being used within the scope of the function, so I don't think it would make it hard to keep track of state changes.

I could just document the function so it mentions that it mutates the root param 🤔 

Another thought I had was to just make `adjustEmbeddedTransactionTimestamps` a method on `SpanTreeModel`, so we can just call `parsedRootSpan.adjustEmbeddedTransactionTimestamps()`, but this may be less memory efficient, since the function would exist on every instance?

I'm curious to know if @dashed has any thoughts on this matter 😄

Reply:
@0Calories An alternative could be to make the clock adjustment on `parsedTrace.childSpans` directly as a for loop, just before `const parsedRootSpan = new SpanTreeModel(...)` on line 774.

This lets you avoid mutating on `SpanTreeModel` instances directly, and be able to retain `span: Readonly<SpanType>;` as a prop on `SpanTreeModel`.

Something like:

```js
parsedTrace.childSpans.forEach(childSpan => {
  childSpan.start_timestamp += startTimeDelta;
  childSpan.timestamp += startTimeDelta;
});
```

Reply:
Sounds good, I'll try to do it this way

Reply:
Added